app/next-client-app/components/data-table/index.tsx

```diff
@@ -91,39 +93,41 @@ export function DataTable<TData, TValue>({
       <div className="flex justify-between mb-3">
         {Filter}
         {/* Views Columns Menu */}
-        <DropdownMenu>
-          <DropdownMenuTrigger asChild>
-            <Button
-              aria-label="Toggle columns"
-              variant="outline"
-              className="ml-auto hidden lg:flex"
-            >
-              <Columns3 className="mr-2 size-4" />
-              Columns
-            </Button>
-          </DropdownMenuTrigger>
-          <DropdownMenuContent align="end">
-            <DropdownMenuLabel>Toggle columns</DropdownMenuLabel>
-            <DropdownMenuSeparator />
-            {table
-              .getAllColumns()
-              .filter((column) => column.getCanHide())
-              .map((column) => {
-                return (
-                  <DropdownMenuCheckboxItem
-                    key={column.id}
-                    className="capitalize"
-                    checked={column.getIsVisible()}
-                    onCheckedChange={(value) =>
-                      column.toggleVisibility(!!value)
-                    }
-                  >
-                    {column.id}
-                  </DropdownMenuCheckboxItem>
-                );
-              })}
-          </DropdownMenuContent>
-        </DropdownMenu>
+        {normalTable && (
```

Comment:
How come this (`normalTable`) is necessary? It being called `normalTable` sounds very generic and therefore there's probably a better way of handling it. Or it needs to be clearer as to why and what it is doing

Reply:
this is because the JobDialog will be using the DataTable component which has the Pagination and ViewColumns components. When there are many DataTable inside JobDialog (because each datatable represents one run), Pagination and ViewCols will be not necessary and make the Dialog content messy, so I thought I would remove them for the JobDialog, using this `normalTable` param

Reply:
Okay thanks I understand, then yes I think you're on to a good approach, but I think we use this as an opportunity to be more explicit about the datatable component.
As in - instead of `normalTable` turning on/off the pagination and viewcols, we could turn those into explicit paramaters.

`paginated` = `true` as default to turn on/off pagination
`viewColumns` = `true` to turn on/off the menu (maybe you have a better name though?)

What do you think?